pyteal/ast/scratchvar.py

```diff
@@ -1,11 +1,30 @@
+from abc import ABC, abstractmethod
+
 from pyteal.errors import TealInputError
-from pyteal.types import TealType, require_type
+from pyteal.types import TealType, require_type, types_match
 
 from pyteal.ast.expr import Expr
 from pyteal.ast.scratch import ScratchSlot, ScratchLoad, ScratchStore
 
 
-class ScratchVar:
+class AbstractVar(ABC):
```

Comment:
Nit: since `AbstractVar` may (and in fact _is_) subclassed by non-scratch related classes, I recommend a new file `abstractvar.py` to contain its source.

Reply:
that's also good point, do we need a set of test for that? I suppose no, for it is pure abstract class.